### PR TITLE
Build server layer endpoints_CRUD Routes

### PR DESCRIPTION
**Pull request by clairesujin8702**
Please review and confirm merge!
Thanks

### CODE REVIEW

File: server/routers/adRouter.js [hunk 1]
```diff
@@ -1,9 +1,30 @@
...
+
+router.get('/rooms', controller.rooms.get);
+router.post('/rooms', controller.rooms.post);
+router.put('/rooms', controller.rooms.put);
+router.put('/rooms/:reservationId', controller.rooms.put);
+router.delete('/rooms', controller.rooms.delete);
+
+router.get('/tasks', controller.tasks.get);
+router.post('/tasks', controller.tasks.post);
+router.put('/tasks', controller.tasks.put);
```
Comment (Orenjiku) [inline server/routers/adRouter.js]: Change tasks PUT route from '/tasks' to '/tasks/:task_id'.

File: server/routers/adRouter.js [hunk 2]
```diff
@@ -1,9 +1,30 @@
...
+router.put('/tasks/:roomId', controller.tasks.put);
```
Comment (Orenjiku) [inline server/routers/adRouter.js]: Remove '/tasks/:roomId' PUT route. The route to '/tasks/:task_id' can handle multiple parameters.

File: server/routers/adRouter.js [hunk 3]
```diff
@@ -1,9 +1,30 @@
...
+router.delete('/tasks', controller.tasks.delete);
```
Comment (Orenjiku) [inline server/routers/adRouter.js:20]: No delete routing expected for in MVP. Can remove.

File: server/routers/fdRouter.js [hunk 4]
```diff
@@ -1,9 +1,28 @@
...
+router.put('/rooms/:reservationId', controller.rooms.put);
+router.delete('/rooms', controller.rooms.delete);
+
+router.get('/tasks', controller.tasks.get);
+router.post('/tasks', controller.tasks.post);
+router.put('/tasks', controller.tasks.put);
+router.put('/tasks/:roomId', controller.tasks.put);
+router.delete('/tasks', controller.tasks.delete);
+
+router.get('/employees', controller.employees.get);
```
Comment (Orenjiku) [inline server/routers/fdRouter.js:22]: Need a route for '/employees/:employee_id'

File: server/routers/adRouter.js [hunk 5]
```diff
@@ -1,9 +1,30 @@
...
+
+router.get('/employees', controller.employees.get);
```
Comment (Orenjiku) [inline server/routers/adRouter.js:24]: Add a GET route for '/employees/:employee_id'

File: server/routers/adRouter.js [hunk 6]
```diff
@@ -1,9 +1,30 @@
...
+router.post('/employees', controller.employees.post);
+router.put('/employees', controller.employees.put);
```
Comment (Orenjiku) [inline server/routers/adRouter.js]: Can remove PUT route to '/employees'. Updating employees can be handled with '/employees/:employeeId' PUT route below.

File: server/routers/adRouter.js [hunk 7]
```diff
@@ -1,9 +1,30 @@
...
-router.delete('/', controller.admin.delete);
+router.get('/reservations', controller.reservations.get);
+router.post('/reservations', controller.reservations.post);
+router.put('/reservations', controller.reservations.put);
+router.put('/reservations/:guestId', controller.reservations.put);
+router.delete('/reservations', controller.reservations.delete);
+
+router.get('/rooms', controller.rooms.get);
+router.post('/rooms', controller.rooms.post);
+router.put('/rooms', controller.rooms.put);
```
Comment (Orenjiku) [inline server/routers/adRouter.js]: Can remove PUT route to '/rooms'. Room updates will be handled by '/rooms/:room_id'

File: server/routers/adRouter.js [hunk 8]
```diff
@@ -1,9 +1,30 @@
...
+router.put('/rooms/:reservationId', controller.rooms.put);
```
Comment (Orenjiku) [inline server/routers/adRouter.js]: Change PUT route from '/rooms/:reservationId' to '/rooms/:room_id'

File: server/routers/adRouter.js [hunk 9]
```diff
@@ -1,9 +1,30 @@
...
-router.post('/', controller.admin.post);
-router.put('/', controller.admin.put);
-router.delete('/', controller.admin.delete);
+router.get('/reservations', controller.reservations.get);
+router.post('/reservations', controller.reservations.post);
+router.put('/reservations', controller.reservations.put);
+router.put('/reservations/:guestId', controller.reservations.put);
+router.delete('/reservations', controller.reservations.delete);
+
+router.get('/rooms', controller.rooms.get);
```
Comment (Orenjiku) [inline server/routers/adRouter.js:11]: Add a GET route to '/rooms/:room_id'
Add a GET route to '/rooms/amenities'
Add a GET route to '/rooms/types'

File: server/routers/adRouter.js [hunk 10]
```diff
@@ -1,9 +1,30 @@
 const router = require('express').Router();
 const controller = require('../controllers');
 
-router.get('/', controller.admin.get);
-router.post('/', controller.admin.post);
-router.put('/', controller.admin.put);
-router.delete('/', controller.admin.delete);
+router.get('/reservations', controller.reservations.get);
```
Comment (Orenjiku) [inline server/routers/adRouter.js:4]: Add GET route '/reservations/availability/:date'

File: server/routers/adRouter.js [hunk 11]
```diff
@@ -1,9 +1,30 @@
...
+router.post('/reservations', controller.reservations.post);
+router.put('/reservations', controller.reservations.put);
```
Comment (Orenjiku) [inline server/routers/adRouter.js]: Change PUT route from '/reservations' to '/reservations/checkIn/reservation_id'

File: server/routers/adRouter.js [hunk 12]
```diff
@@ -1,9 +1,30 @@
...
+router.put('/reservations/:guestId', controller.reservations.put);
```
Comment (Orenjiku) [inline server/routers/adRouter.js]: Change PUT route from '/reservations/:guestId' to 'reservations/checkOut/:reservations_id'

**Review by Orenjiku — COMMENTED** [review]
Colin added a FlamingoAPI.md file that lists out the routes that we will be using. I added some comments for the changes. But you can follow that document for the routes.

File: server/routers/adRouter.js (hunk 6 above)
Comment (clairesujin8702) [reply server/routers/adRouter.js]: To make it clear, PUT route is using "**employeeId**" or "**employee_id**" ?

Comment (clairesujin8702) [reply server/routers/adRouter.js]: I saw your comment below i will check FlamingoAPI.md file as well. 
Thank you for reviewing!!

**Pull request closed without merging**